### ISSUE TITLE
[Feature] Waveform focus styles

### ISSUE THREAD

**Issue by sarayourfriend** [WordPress/openverse-frontend#147]
## Problem
The waveform does not have a clear focus style, meaning that sighted keyboard users will have a difficult time knowing that they are able to seek using the keyboard like other "seek bars" that exist.

## Description
We should add a focus style that makes it clear to sighted keyboard users that they can seek using the arrow keys. Screen reader users will get an indiciation from their screen reader that they are in a "slider" which will give the hint that they can use the arrow keys to interact, but for non-screen reader keyboard users this will not apply. Therefore we need some kind of visual indication that the waveform is focused and seekable.

One potential solution is to make the vertical "seek" line a solid color with a circle or bulb at the top. Similar to how iOS represents a video scub in the Photos app, but with a bulb at the top of the line to add an extra indication (especially when the audio is at the `0:00` timestamp something will need to be visible inside of the waveform).

![IMG_1064](https://user-images.githubusercontent.com/24264157/130517094-e7c467f0-973d-4400-a5c4-56726cec20c7.jpg)

## Alternatives
I'm sure there are other designs we could do for the focus styles, but I'm not sure of any.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

**Comment by zackkrida** [comment]
@panchovm This needs some small design thinking applied, please!

**Comment by panchovm** [comment]
Yes. I'm working this week on this

**Comment by panchovm** [comment]
I am trying different styles, but something that comes to my mind is how the interaction works.

Once the focus is on the waveform and arrow keys go forward/backward, does it play the sound immediately? Or does it need a second action (like space bar) to start playing?

On youtube, it is tied to the previous state. If the video is playing, arrow keys go 5 secs forward/backward and the video plays immediately. If the video is paused, it does not play until pressing the space bar. Same on SoundCloud and Spotify.

Last.fm allows playing/pause with the space bar but not moving across time with arrow keys. While Europeana, Jamendo, and Wikimedia Commons do not allow navigation on the audio time.

### Youtube
<img src="https://user-images.githubusercontent.com/895819/131569195-7291164f-28e1-4cc8-80a1-2a100ae7df09.gif" width="600"/>

### Soundcloud
<img src="https://user-images.githubusercontent.com/895819/131569431-a2f4b76e-b15e-4123-bccd-469004248d20.gif" width="600"/>

### Spotify web
<img src="https://user-images.githubusercontent.com/895819/131569493-ef5a25a4-4eb7-4436-97e1-6f3de4cf2960.gif" width="600"/>

**Comment by sarayourfriend** [comment]
It works the same as YouTube and Spotify 🙂 Which I think is the most sensible. If you're trying to seek to a specific time before playing the audio, it would be really surprising to me if it suddenly started playing as well. In any case, if seeking using the keyboard we'd have to debounce the auto-playing behavior for some arbitrary amount of time.

**Comment by panchovm** [comment]
Right. That makes sense.

Now I wonder how useful is the 1-sec navigation feature, especially on long audios. I agree with the idea of missing specific seconds, but I guess that seeking time 01:13 but arriving and playing from 01:10 is not annoying.

Youtube iOS has jump periods of 10, 20, 30, and 40 seconds when tapping the video area to go forward and backward. I wonder how that behavior might feel in this case.

**Comment by zackkrida** [comment]
@panchovm we have a modifier for `shift + ->` and `shift + <-` to jump by 15 seconds. I'll get you instructions for how to run the site locally this week too, i forgot about that!

**Comment by panchovm** [comment]
You are right. We decided that and I forgot it. Thanks for bringing it back.

**Comment by panchovm** [comment]
Here is the version I like most.

![Keyboard navigation](https://user-images.githubusercontent.com/895819/131724626-652dd061-ce64-4c79-92d1-1fe3b4821d18.png)

Once the focus is on the wave area, the focus mark is visible and, depending on the audio state (play or pause), it behaves as indicated above. I would like to see how it feels when interacting in real since the mockup is a static depiction.

I tried other styles but did not like them much, yet attached below.

![Kayboard navigation  Focus styles](https://user-images.githubusercontent.com/895819/131726725-5ab5bf27-58b2-4897-858f-3a3b1e0d7f7f.png)

## Audio component parts

While designing the component, I faced many unnamed parts that might be a good idea to call them somehow. That is why I made this draft to align us with all concepts used from now. Indeed, this is fully open to changes and enhancements.

<img width="1378" alt="Screen Shot 2021-09-01 at 14 49 31" src="https://user-images.githubusercontent.com/895819/131727045-113a5332-b993-45ff-9a4c-ab3ad3ab4d9c.png">

**Comment by panchovm** [comment]
Here is [the last version](https://www.figma.com/file/vEoItpMQyFZb5AZV9aoP7I/Audio-integration?node-id=991%3A12336) of audio component with all states.

**Comment by panchovm** [comment]
I added a mockup with focus styles per area. This is to highlight the focus area and being more explicit. In that vein, the **focus mark** should appear once the focus is in the wave area.

<img width="1225" alt="Screen Shot 2021-09-06 at 10 50 23" src="https://user-images.githubusercontent.com/895819/132227548-172ad8ab-7b1d-46dc-86ef-a88e7738c87f.png">

**Comment by sarayourfriend** [comment]
@panchovm that's great! I was going to comment (but forgot to) on the PR that it was hard to notice that the area was focused at all without an additional outline.
